### PR TITLE
Feature/video indexing skill

### PR DESCRIPTION
**Pull request by graemefoster**
## Purpose
Adds a skill which can invoke the Azure Video Indexing endpoint, allowing video content to be indexed and accessible through search.

## Does this introduce a breaking change?
```
[ ] Yes
[x ] No
```

## Pull Request Type
```
[ ] Bugfix
[x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone https://github.com/graemefoster/azure-search-power-skills
cd azure-search-power-skills
git checkout feature/video-indexing-skill
npm install
```

* Test the code

```
cd .\Tests\PowerSkillTests
dotnet test

```

## What to Check
Verify that the following are valid
* VideoIndexerTests

## Other Information
I've added fairly detailed instructions in the readme.md on how to use the skill. I've been using the skill by daisy chaining it with another indexer that picks up the simplified video-indexer insights, and appends it to the index model in the Cognitive Search Knowledge Mining sample.

### CODE REVIEW

File: Video/VideoIndexer/VideoIndexer.csproj [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <AzureFunctionsVersion>v2</AzureFunctionsVersion>
```
Comment (dereklegenzoff) [inline Video/VideoIndexer/VideoIndexer.csproj:4]:
```suggestion
    <AzureFunctionsVersion>v3</AzureFunctionsVersion>
```
There's a mismatch between the csproj file and arm template causing the deployment to fail

File: Video/VideoIndexer/README.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            {
+              "metadata_storage_path": "<SAFE-BASE64-ENCODED-PATH-TO-VIDEO>",
+              "metadata_storage_name": "<VIDEO-NAME>"
+            }
+        }
+    ]
+}
+```
+
+### Sample Output:
```
Comment (dereklegenzoff) [inline Video/VideoIndexer/README.md:58]: Is the output empty by design? If so, could you maybe add a comment here saying it is.

File: Video/VideoIndexer/README.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+| MediaIndexer_StorageConnectionString | Azure Storage Connection string pointing to a blob store to store simplified Video Indexer results |
+| MediaIndexer_StorageContainer | Name of the container to place the simplified Video Indexer results |
+| MediaIndexer_CallbackFunctionCode | Azure Function code enabling the Video Indexer to invoke the callback function |
+
+
+## Deployment
+
+[![Deploy to Azure](https://azuredeploy.net/deploybutton.svg)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fgraemefoster%2Fazure-search-power-skills%2Ffeature%2Fvideo-indexing-skill%2FVideo%2FVideoIndexer%2Fazuredeploy.json)
+
+## hello-world
```
Comment (dereklegenzoff) [inline Video/VideoIndexer/README.md:39]:
```suggestion
## Video Indexer
```

File: Video/VideoIndexer/README.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            "data": {
+            },
+            "errors": [],
+            "warnings": []
+        }
+    ]
+}
+```
+
+### Sample Insights Blob Output
```
Comment (dereklegenzoff) [inline Video/VideoIndexer/README.md:74]: Could you add a sentence explaining what this is? I'm assuming this is what ends up in blob storage which is then indexed?

File: Video/VideoIndexer/README.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        "name": "metadata_storage_name",
+        "source": "/document/metadata_storage_name"
+      }
+    ],
+    "outputs": [
+    ]
+}
+```
+
+You will also need an indexer that points to the container where simplified insights are placed. Here's an example of that. Notice how in this example we map the ```originalVideoEncodedMetadataPath``` and ```originalVideoName``` which identify the original video index item.
```
Comment (dereklegenzoff) [inline Video/VideoIndexer/README.md:125]: Thanks for including this section -- super valuable.

File: Video/VideoIndexer/README.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+      {
+        "name": "metadata_storage_name",
+        "source": "/document/metadata_storage_name"
+      }
+    ],
+    "outputs": [
+    ]
+}
+```
+
```
Comment (dereklegenzoff) [inline Video/VideoIndexer/README.md:124]:
```suggestion

## Indexing the Video Indexer output

```

Could you add an H2 here to distinguish this section a bit more? It's really valuable so I don't want folks to miss it (I kind of missed it when I read through)

File: Video/VideoIndexer/VideoIndexer.csproj (hunk 1 above)
Comment (graemefoster) [reply Video/VideoIndexer/VideoIndexer.csproj:4]: oops that's embarrassing! Last change I made after I deployed it 😊

Comment (graemefoster) [reply Video/VideoIndexer/VideoIndexer.csproj]: Fixed

File: Video/VideoIndexer/README.md (hunk 2 above)
Comment (graemefoster) [reply Video/VideoIndexer/README.md:58]: Added better documentation

File: Video/VideoIndexer/README.md (hunk 3 above)
Comment (graemefoster) [reply Video/VideoIndexer/README.md]: Added better documentation

File: Video/VideoIndexer/README.md (hunk 4 above)
Comment (graemefoster) [reply Video/VideoIndexer/README.md:78]: Added better documentation

File: Video/VideoIndexer/README.md (hunk 5 above)
Comment (graemefoster) [reply Video/VideoIndexer/README.md]: 👍

File: Video/VideoIndexer/README.md (hunk 6 above)
Comment (graemefoster) [reply Video/VideoIndexer/README.md:130]: Added better documentation

File: Video/VideoIndexer/README.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+| Setting Name | Description |
+| ---- | --- |
+| MediaIndexer_AccountId | Azure Video Indexer Account Id |
+| MediaIndexer_Location | Azure Video Indexer Account location, e.g. trial |
+| MediaIndexer_AccountKey | Azure Video Indexer Account key |
+| MediaIndexer_StorageConnectionString | Azure Storage Connection string pointing to a blob store to store simplified Video Indexer results |
+| MediaIndexer_StorageContainer | Name of the container to place the simplified Video Indexer results |
+| MediaIndexer_CallbackFunctionCode | Azure Function code enabling the Video Indexer to invoke the callback function |
+
+
```
Comment (dereklegenzoff) [inline Video/VideoIndexer/README.md:34]:
```suggestion
To get started with video indexer and to get the necessary AccountId and AccountKey, you can follow this [tutorial](https://docs.microsoft.com/azure/media-services/video-indexer/video-indexer-use-apis).

```

File: Video/VideoIndexer/README.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+    ],
+    "outputs": [
+    ]
+}
+```
+
+## Indexing the output of the Video Indexer
+
+Finally, you will need an indexer that indexes the contents of the container where simplified insights are placed. Here's an example of that. Notice how in this example we map the ```originalVideoEncodedMetadataPath``` and ```originalVideoName``` which identify the original video index item. 
+
```
Comment (dereklegenzoff) [inline Video/VideoIndexer/README.md:139]:
```suggestion

> NOTE: By default, this skill will output the video contents to the storage account deployed as part of this ARM template so you will need to create a new data source as well.

```

File: Video/VideoIndexer/README.md [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+| MediaIndexer_Location | Azure Video Indexer Account location, e.g. trial |
+| MediaIndexer_AccountKey | Azure Video Indexer Account key |
+| MediaIndexer_StorageConnectionString | Azure Storage Connection string pointing to a blob store to store simplified Video Indexer results |
+| MediaIndexer_StorageContainer | Name of the container to place the simplified Video Indexer results |
+| MediaIndexer_CallbackFunctionCode | Azure Function code enabling the Video Indexer to invoke the callback function |
+
+
+## Deployment
+
+[![Deploy to Azure](https://azuredeploy.net/deploybutton.svg)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fgraemefoster%2Fazure-search-power-skills%2Ffeature%2Fvideo-indexing-skill%2FVideo%2FVideoIndexer%2Fazuredeploy.json)
```
Comment (dereklegenzoff) [inline Video/VideoIndexer/README.md:37]:
```suggestion
[![Deploy to Azure](https://azuredeploy.net/deploybutton.svg)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure-Samples%2Fazure-search-power-skills%2Fmaster%2FVideo%2FVideoIndexer%2Fazuredeploy.json)
```

**Pull request merged**